doc/internal/man3/evp_keymgmt_freekey.pod

```diff
@@ -0,0 +1,137 @@
+=pod
+
+=head1 NAME
+
+evp_keymgmt_importdomain, evp_keymgmt_gendomain,
+evp_keymgmt_freedomain,
+evp_keymgmt_exportdomain,
+evp_keymgmt_importdomain_types, evp_keymgmt_exportdomain_types,
+evp_keymgmt_importkey_priv, evp_keymgmt_importkey_pub, evp_keymgmt_genkey,
+evp_keymgmt_loadkey,
+evp_keymgmt_freekey,
+evp_keymgmt_exportkey_priv, evp_keymgmt_exportkey_pub,
+evp_keymgmt_importkey_priv_types, evp_keymgmt_importkey_pub_types,
+evp_keymgmt_exportkey_priv_types, evp_keymgmt_exportkey_pub_types
+- internal KEYMGMT support functions
+
+=head1 SYNOPSIS
+
+ #include "internal/evp_int.h"
+
+ void *evp_keymgmt_importdomain(const EVP_KEYMGMT *keymgmt,
+                                const OSSL_PARAM params[]);
+ void *evp_keymgmt_gendomain(const EVP_KEYMGMT *keymgmt,
+                             const OSSL_PARAM params[]);
+ void evp_keymgmt_freedomain(const EVP_KEYMGMT *keymgmt, void *provdomain);
+ int evp_keymgmt_exportdomain(const EVP_KEYMGMT *keymgmt,
+                              void *provdomain, OSSL_PARAM params[]);
+ const OSSL_PARAM *evp_keymgmt_importdomain_types(const EVP_KEYMGMT *keymgmt);
+ const OSSL_PARAM *evp_keymgmt_exportdomain_types(const EVP_KEYMGMT *keymgmt);
+ 
+ void *evp_keymgmt_importkey_priv(const EVP_KEYMGMT *keymgmt,
+                                  const OSSL_PARAM params[]);
+ void *evp_keymgmt_importkey_pub(const EVP_KEYMGMT *keymgmt,
+                                const OSSL_PARAM params[]);
+ void *evp_keymgmt_genkey(const EVP_KEYMGMT *keymgmt, void *domain,
+                          const OSSL_PARAM params[]);
+ void *evp_keymgmt_loadkey(const EVP_KEYMGMT *keymgmt,
+                           void *id, size_t idlen);
+ void evp_keymgmt_freekey(const EVP_KEYMGMT *keymgmt, void *provkey);
+ int evp_keymgmt_exportkey_priv(const EVP_KEYMGMT *keymgmt,
+                                void *provkey, OSSL_PARAM params[]);
+ int evp_keymgmt_exportkey_pub(const EVP_KEYMGMT *keymgmt,
+                               void *provkey, OSSL_PARAM params[]);
+ const OSSL_PARAM *
+ evp_keymgmt_importkey_priv_types(const EVP_KEYMGMT *keymgmt);
+ const OSSL_PARAM *
+ evp_keymgmt_importkey_pub_types(const EVP_KEYMGMT *keymgmt);
+ const OSSL_PARAM *
+ evp_keymgmt_exportkey_priv_types(const EVP_KEYMGMT *keymgmt);
+ const OSSL_PARAM *
+ evp_keymgmt_exportkey_pub_types(const EVP_KEYMGMT *keymgmt);
+
+=head1 DESCRIPTION
+
+All these functions are helpers to call the provider's corresponding
+function.
+
+evp_keymgmt_importdomain() calls the method's importdomain() function.
```

Comment:
Should we be documenting what "importdomain" actually *does*? Similarly with the other lines below.

Reply:
Yeah, we should.  I *also* think we should document the dispatch functions, I just dunno how, or rather where.  (I'm trying to document in the header file at the very least, something we generally suck at, I might add...)

Reply:
Whether in this PR or another, we're going to need to document all of these provider functions in the man pages. A provider author should be able to find all the information they need in the man pages, without have to read the source.

Reply:
Agreed.  For the `core_numbers.h` stuff, that will be another PR.  I have ideas on that...